spec/controllers/v0/user_action_events_controller_spec.rb

```diff
@@ -0,0 +1,201 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe V0::UserActionEventsController, type: :controller do
+  include RequestHelper
+
+  context 'when not logged in' do
```

Comment:
This should get moved into the `describe 'GET #index'`  block since it is behavior of the `index` call that is being tested.